apps/archive/archive.py

```diff
@@ -222,6 +223,21 @@ def on_update(self, updates, original):
         if original[ITEM_TYPE] == CONTENT_TYPE.PICTURE:  # create crops
             CropService().create_multiple_crops(updates, original)
 
+        updates_feature_image = updates.get('associations', {}).get('feature_image')
+        if updates_feature_image and 'poi' in updates_feature_image:
+            original_feature_image = original.get('associations', {}).get('feature_image', {})
+            if original_feature_image and original_feature_image.get('poi', {}) == updates_feature_image['poi']:
+                return
+            _id = updates_feature_image[config.ID_FIELD] if config.ID_FIELD in updates_feature_image \
+                    else original_feature_image[config.ID_FIELD]
+            archive_service = get_resource_service(ARCHIVE)
+            image_item = archive_service.find_one(req=None, _id=_id)
```

Comment:
a minor one: `get_resource_service(ARCHIVE)` not needed. Class is Archive service.